extract-operator-bundle.yml

```diff
@@ -0,0 +1,22 @@
+---
+- name: Extract and parse the operator bundle image for testing usage
+  hosts: all
+  become: false
+  gather_facts: false
+
+  vars:
+    work_dir: "/tmp/operator-test"
+    operator_work_dir: "{{ work_dir }}/operator-files"
+    operator_bundle_dir: "{{ work_dir }}/operator-bundle"
+    testing_bin_path: "{{ work_dir }}/bin"
+    jq_bin_path: "{{ testing_bin_path }}/jq"
+    yq_bin_path: "{{ testing_bin_path }}/yq"
+
+  tasks:
+    - name: "Extract the operator bundle image into files needed to run the tests"
+      include_role:
+        name: extract_operator_bundle
```

Comment:
Do We want to simulate something like "atomic mount" for ostree format?
If yes, then much better is to use already existing tools for that.

The atomic utility is removed from fedora > 30 so we needn't rely on ostree format.
But we can use `oci` format with `umoci` utility
https://github.com/openSUSE/umoci
`umouci unpack`
e.g.

```
sh-5.0# skopeo copy docker://some.docker.registry/bundle:v1.1-15 oci:/tmp/data/oci:latest
Getting image source signatures
Copying blob 07aec32d6429 [--------------------------------------] 0.0b / 0.0b
Copying config 1e79bfc95f [======================================] 1.6KiB / 1.6KiB
Writing manifest to image destination
Storing signatures
sh-5.0# skopeo inspect --raw oci:/tmp/data/oci:latest
{"schemaVersion":2,"config":{"mediaType":"application/vnd.oci.image.config.v1+json","digest":"sha256:1e79bfc95fd787192af6ac84e95735f80c7506bf1a1e973bb5a5805b75bb92d1","size":1687},"layers":[{"mediaType":"application/vnd.oci.image.layer.v1.tar+gzip","digest":"sha256:07aec32d6429e630c6b4d245eecb35b25a3748f78c5bccd07e061f285dcebf4f","size":7504}]}sh-5.0#
sh-5.0#

sh-5.0# rm -rf /tmp/data/dest_dir
sh-5.0# ./umoci.amd64 unpack --rootless --image /tmp/data/oci:latest /tmp/data/dest_dir

sh-5.0# ls /tmp/data/dest_dir/rootfs/
manifests  metadata  root

sh-5.0# find /tmp/data/dest_dir/
/tmp/data/dest_dir/
/tmp/data/dest_dir/umoci.json
/tmp/data/dest_dir/sha256_70b31a344256c3a097ca5ec82947688208b9c0b5f6afbfebb7f95cd896c03e79.mtree
/tmp/data/dest_dir/config.json
/tmp/data/dest_dir/rootfs
/tmp/data/dest_dir/rootfs/manifests
/tmp/data/dest_dir/rootfs/manifests/volumesnapshotlocation.crd.yaml
/tmp/data/dest_dir/rootfs/manifests/serverstatusrequest.crd.yaml
/tmp/data/dest_dir/rootfs/manifests/schedule.crd.yaml
/tmp/data/dest_dir/rootfs/manifests/restore.crd.yaml
/tmp/data/dest_dir/rootfs/manifests/resticrepository.crd.yaml
/tmp/data/dest_dir/rootfs/manifests/podvolumerestore.crd.yaml
/tmp/data/dest_dir/rootfs/manifests/podvolumebackup.crd.yaml
/tmp/data/dest_dir/rootfs/manifests/migstorage.crd.yaml
/tmp/data/dest_dir/rootfs/manifests/migrationcontroller.crd.yaml
/tmp/data/dest_dir/rootfs/manifests/migplan.crd.yaml
/tmp/data/dest_dir/rootfs/manifests/migmigration.crd.yaml
/tmp/data/dest_dir/rootfs/manifests/migcluster.crd.yaml
/tmp/data/dest_dir/rootfs/manifests/mig-operator.v1.1.1.clusterserviceversion.yaml
/tmp/data/dest_dir/rootfs/manifests/downloadrequest.crd.yaml
/tmp/data/dest_dir/rootfs/manifests/deletebackuprequest.crd.yaml
/tmp/data/dest_dir/rootfs/manifests/backupstoragelocation.crd.yaml
/tmp/data/dest_dir/rootfs/manifests/backup.crd.yaml
/tmp/data/dest_dir/rootfs/metadata
/tmp/data/dest_dir/rootfs/metadata/annotations.yaml
/tmp/data/dest_dir/rootfs/root
/tmp/data/dest_dir/rootfs/root/buildinfo
/tmp/data/dest_dir/rootfs/root/buildinfo/Dockerfile-bundle-v1.1-15
```

Reply:
That looks like a good idea, will try it out.

Reply:
Switched the unpacking method to use umoci, please take a look.

Reply:
ty, it looks much simpler :-)